source/file.go

```diff
@@ -126,11 +126,14 @@ func NewFile(relpath string) *File {
 	f.ext = strings.TrimPrefix(filepath.Ext(f.LogicalName()), ".")
 	f.baseName = helpers.Filename(f.LogicalName())
 
-	f.lang = strings.TrimPrefix(filepath.Ext(f.baseName), ".")
-	if f.lang == "" {
+	lang := strings.TrimPrefix(filepath.Ext(f.baseName), ".")
+	if _, ok := viper.GetStringMap("Languages")[lang]; lang == "" || !ok {
```

Comment:
All the rest of viper.Get... in Hugo now camelCase keys, so we might as well be pedantic about it. Other than that this looks good.